### ISSUE TITLE
Handle avatar-URL optimization `user_avatar_url_field_optional`

### ISSUE THREAD

**Issue by gnprice** [zulip/zulip-flutter#254]
That is, the `user_avatar_url_field_optional` client capability in the Zulip API:
https://zulip.com/api/register-queue#parameter-client_capabilities

At the moment, we claim that capability in our registerQueue requests; but we don't actually implement what we need on our side in order to handle it, so we end up with missing avatars. We'll shortly flip that so that we no longer claim the capability, getting correct behavior. But then we should actually implement the logic (there's not much to it), and re-enable the optimization.

For how the implementation can work, see `FallbackAvatarURL` in zulip-mobile's [src/utils/avatar.js](https://github.com/zulip/zulip-mobile/blob/c979530d6804db33310ed7d14a4ac62017432944/src/utils/avatar.js). In particular see `validateAndConstructInstance` and `get`. I think we can probably do the implementation with less code than is in zulip-mobile, though.

At the moment our `User.avatarUrl` field gets deserialized in a way that doesn't distinguish whether the server left `avatar_url` out or set it to null. We'll need to distinguish those in order to handle `client_gravatar` (a separate issue)… but the fallback avatar URLs should work for all users. So for this issue we should be fine without that distinction.

**Comment by gnprice** [comment]
> We'll shortly flip that so that we no longer claim the capability, getting correct behavior.

(This was done in caa9603784e1a7a7aee0af4682553eed6b1d0033 in #249. Implementing the optimization, which is what the present issue is about, is still open.)

**Comment by NishchayRajput** [comment]
Hi, I noticed that this issue is mentioned inside Zulip Flutter. Should the fix be implemented in Zulip Flutter or Zulip Mobile?
Since the reference code (FallbackAvatarURL) is from Zulip Mobile, should we replicate a similar implementation in Zulip Flutter, or does it need changes in both apps?

**Comment by Abdelmohaymn** [comment]
Hi! I would love to pick this up. I've reviewed the **FallbackAvatarURL** implementation in **zulip-mobile/src/utils/avatar.js**. 
I plan to port this logic to Dart by constructing the **/avatar/{user_id} (and /medium)** URLs when the **avatar_url** is absent. 
Once the fallback generation is in place, I will re-enable the **user_avatar_url_field_optional** capability in the register request. I'll also add the necessary tests. Can I go ahead and claim this?
